1.5.0-beta01 CHANGELOG (#3407)
JetBrains/compose-multiplatform: CHANGELOG.md

# 1.5.0-beta01 (July 2023)

## Common

### Features

> - [Dialog in common](https://github.com/JetBrains/compose-multiplatform-core/pull/632)
> - [Popup in common](https://github.com/JetBrains/compose-multiplatform-core/pull/611)

Reviewer: ```suggestion
- [`Dialog` API is available to use from common source set](https://github.com/JetBrains/compose-multiplatform-core/pull/632)
- [`Popup` API is available to use from common source set](https://github.com/JetBrains/compose-multiplatform-core/pull/611)
```

> - [Insets in common](https://github.com/JetBrains/compose-multiplatform-core/pull/586)

Reviewer: ```suggestion
- [`WindowInsets` API is available to use from common source set](https://github.com/JetBrains/compose-multiplatform-core/pull/586)
```

> - [A warning if compose.kotlinCompilerPlugin is set to androidx.compose.compiler.compiler](https://github.com/JetBrains/compose-multiplatform/pull/3313)

Reviewer: ```suggestion
- [A warning if `compose.kotlinCompilerPlugin` is set to `androidx.compose.compiler.compiler`](https://github.com/JetBrains/compose-multiplatform/pull/3313)
```

- [Simplify resource management](https://github.com/JetBrains/compose-multiplatform/pull/3340) (works without CocoaPods now)
- [TextField keyboardOptions, capitalization](https://github.com/JetBrains/compose-multiplatform/issues/2735)
- [TextField, keyboard behavior when have fullscreen size TextField](https://github.com/JetBrains/compose-multiplatform/issues/2752)
- [TextField, different behavior strategies on appearing the keyboard](https://github.com/JetBrains/compose-multiplatform/issues/3128)
- [Insets on iOS](https://github.com/JetBrains/compose-multiplatform-core/pull/577)
- [Support Density.textSize (Dynamic Type)](https://github.com/JetBrains/compose-multiplatform/issues/2567)
- [Change default fonts](https://github.com/JetBrains/compose-multiplatform-core/pull/552) (San Francisco is the default font)
- [UIKit public LocalUIViewController](https://github.com/JetBrains/compose-multiplatform-core/pull/501)

### Fixes
- [Fix dynamic framework support](https://github.com/JetBrains/skiko/pull/763)
- [Fiz TextField context menu](https://github.com/JetBrains/compose-multiplatform/issues/3276)
- [Fix complex blending on iOS](https://github.com/JetBrains/skiko/pull/728)

> - [ViewConfiguration.touchSlop value is quite low on iOS](https://github.com/JetBrains/compose-multiplatform/issues/3397)
> - [Fix topLeftOffset calculation on iOS in Split View](https://github.com/JetBrains/compose-multiplatform-core/pull/678)
> - [Modifier.draggable onDragStopped not called](https://github.com/JetBrains/compose-multiplatform/issues/3310)
> - [UIKitView. Fix lifetime discrepancy within the composition](https://github.com/JetBrains/compose-multiplatform-core/pull/576/files)
> - [Allow higher than 60hz framerates on CADisplayLink on iPhones](https://github.com/JetBrains/compose-multiplatform-ios-android-template/pull/17)

Reviewer: ```suggestion
- [`ViewConfiguration.touchSlop` value is quite low on iOS](https://github.com/JetBrains/compose-multiplatform/issues/3397)
- [Fix `topLeftOffset` calculation on iOS in Split View](https://github.com/JetBrains/compose-multiplatform-core/pull/678)
- [`Modifier.draggable` `onDragStopped` not called](https://github.com/JetBrains/compose-multiplatform/issues/3310)
- [`UIKitView`. Fix lifetime discrepancy within the composition](https://github.com/JetBrains/compose-multiplatform-core/pull/576/files)
- [Allow higher than 60hz framerates on `CADisplayLink` on iPhones](https://github.com/JetBrains/compose-multiplatform-ios-android-template/pull/17)
```

- Implemented all low-level functions to avoid random crashes
  - [Implement TreeSet](https://github.com/JetBrains/compose-multiplatform/issues/2878)
  - [Implement AnnotatedString.transform](https://github.com/JetBrains/compose-multiplatform-core/pull/523)
  - [Implement WeakHashMap](https://github.com/JetBrains/compose-multiplatform/issues/2877)
  - [Implement typefacesCache](https://github.com/JetBrains/compose-multiplatform/issues/2873)
  - [Implement ExpireAfterAccessCache](https://github.com/JetBrains/compose-multiplatform/issues/2871)
  - [Implement NativeStringDelegate](https://github.com/JetBrains/compose-multiplatform/issues/2876)

### API Changes
 - Resource management was reimplemented. Follow [the guide in the PR](https://github.com/JetBrains/compose-multiplatform/pull/3340) to support new feautures

## Desktop

### Features
- [Swing interop. Experimental off-screen rendering on graphics](https://github.com/JetBrains/compose-multiplatform-core/pull/601)
- [Swing interop. ComposePanel that can be disposed manually](https://github.com/JetBrains/compose-multiplatform-core/pull/620)

> - [Add semantic properties to DialogWindow, Popup and Dialog](https://github.com/JetBrains/compose-multiplatform-core/pull/698)

Reviewer: ```suggestion
- [Add semantic properties to `DialogWindow`, `Popup` and `Dialog`](https://github.com/JetBrains/compose-multiplatform-core/pull/698)
```

> - [Use Segoe UI on Windows by default](https://github.com/JetBrains/compose-multiplatform-core/pull/557)

Reviewer: ```suggestion
- [Use `Segoe UI` as sans serif font on Windows](https://github.com/JetBrains/compose-multiplatform-core/pull/557)
```

### Fixes
- [Fix loosing frames on macOs](https://github.com/JetBrains/skiko/pull/753)

> - [Scrolling LazyColumn via mouse wheel stops working](https://github.com/JetBrains/compose-multiplatform/issues/3366)

Reviewer: ```suggestion
- [Scrolling `LazyColumn` via mouse wheel stops working](https://github.com/JetBrains/compose-multiplatform/issues/3366)
```

- [Fix Could not infer Java runtime version for Java home directory](https://github.com/JetBrains/compose-multiplatform/issues/3133)
- [Usage of deprecated forEachGesture in Scrollbar.desktop.kt](https://github.com/JetBrains/compose-multiplatform/issues/3045)
- Test framework fixes
  - [Implement idling resources for tests](https://github.com/JetBrains/compose-multiplatform-core/pull/599)
  - [Implement `SemanticsNodeInteraction.captureToImage()`](https://github.com/JetBrains/compose-multiplatform-core/pull/589)
  - [Wait until compose is idle before returning from `DesktopTestOwner.getRoots()`](https://github.com/JetBrains/compose-multiplatform-core/pull/588)
  - [When the test main clock is advanced, call `ComposeScene.render` with the current test time](https://github.com/JetBrains/compose-multiplatform-core/pull/584)
  - [Add timestamps to batched test input events, and advance the test clock accordingly when sending them](https://github.com/JetBrains/compose-multiplatform-core/pull/578)
  - [Tests support semanatic nodes inside `Window` and `DialogWindow`](https://github.com/JetBrains/compose-multiplatform-core/pull/697)

### API Changes

> - [Rename desktop's `Dialog` to `DialogWindow`](https://github.com/JetBrains/compose-multiplatform-core/pull/661)

Reviewer: Let's mention changing default Esc behaviour (https://github.com/JetBrains/compose-multiplatform-core/pull/712)

Author: I mentioned it for Popup:
```
[All `Popup` overloads call dismiss on `Esc` key by default](https://github.com/JetBrains/compose-multiplatform-core/pull/712)
```
Didn't mention it for `Dialog` (we didn't have it before) and for `DialogWindow` (we don't close it on Esc, I assume)

## Web

### Features
- [Let `ComposeWindow` accept a custom canvas id (html canvas element id)](https://github.com/JetBrains/compose-multiplatform-core/pull/626)
- [Support `Scrollbar`](https://github.com/JetBrains/compose-multiplatform-core/pull/571)

### Fixes
- [Workaround `ComposeWindow` multiple event listeners on resize](https://github.com/JetBrains/compose-multiplatform-core/pull/692)
- [Fix `Modifier.pointerHoverIcon` for browser apps](https://github.com/JetBrains/compose-multiplatform-core/pull/629)
- [Fux `RoundedCornerShape` with zero corners](https://github.com/JetBrains/compose-multiplatform/issues/3013)
- [Fix `ScrollConfig` in web target](https://github.com/JetBrains/compose-multiplatform-core/pull/628)
- Implemented low-level functions to avoid random crashes:
  - [Implement cache key in PlatformFont](https://github.com/JetBrains/compose-multiplatform-core/pull/671)
  - [implement platform detection to apply correct KeyMapping and FontFamiliesMapping](https://github.com/JetBrains/compose-multiplatform-core/pull/637)
  - [Implement PlatformLocale and StringDelegate for k/wasm](https://github.com/JetBrains/compose-multiplatform-core/pull/625)

>   - [Add missing getAndIncrement in AtomicLong implementation for web](https://github.com/JetBrains/compose-multiplatform-core/pull/570)

Reviewer: Please use ` for code like function and class names

Author: Done
